### ISSUE TITLE
Fix S2589 FP: Simple switch block leads to multiple FPs

### ISSUE THREAD

**Issue by drieseng** [SonarSource/sonar-dotnet#8262]
### Description

S2589 is mistakenly reported for several case entries in a switch block.

### Repro steps

```cs
using System;

public static class Program
{
    private static readonly int Rounds = 23;

    private static void Main()
    {
        Run();
    }

    private static void Run()
    {
        for (var i = 1; i <= Rounds; i++)
        {
            switch (i)
            {
                case 1:
                case 4:
                case 7:
                case 10:
                case 13:
                case 16:
                    Console.WriteLine(i);
                    break;
                case 2:
                case 5:
                case 8:
                case 11:
                case 14:
                    Console.WriteLine(i);
                    break;
                case 3:
                case 6:
                case 9:
                case 12:
                case 15:
                    Console.WriteLine(i);
                    break;
                default:
                    break;
            }
        }
    }
}
```
### Expected behavior

S2589 is not reported.

### Actual behavior

The following diagnostics are reported:

Program.cs(28,22,28,23): error S2583: Change this condition so that it does not always evaluate to 'True'. Some code paths are unreachable. (https://rules.sonarsource.com/csharp/RSPEC-2583)
Program.cs(21,22,21,23): error S2589: Change this condition so that it does not always evaluate to 'False'. (https://rules.sonarsource.com/csharp/RSPEC-2589)
Program.cs(22,22,22,23): error S2589: Change this condition so that it does not always evaluate to 'False'. (https://rules.sonarsource.com/csharp/RSPEC-2589)
Program.cs(23,22,23,24): error S2589: Change this condition so that it does not always evaluate to 'False'. (https://rules.sonarsource.com/csharp/RSPEC-2589)
Program.cs(24,22,24,24): error S2589: Change this condition so that it does not always evaluate to 'False'. (https://rules.sonarsource.com/csharp/RSPEC-2589)
Program.cs(25,22,25,24): error S2589: Change this condition so that it does not always evaluate to 'False'. (https://rules.sonarsource.com/csharp/RSPEC-2589)Please provide a description of the actual behavior you observe.

### Known workarounds

None.

### Related information

* SonarAnalyzer.CSharp version: 9.12.0.78982
* Visual Studio version: 17.8.0 Preview 2.0

**Comment by martin-strecker-sonarsource** [comment]
Thank you, @drieseng. I can confirm the TPs and added a reproducer in #8270. The root cause is probably caused by the complex code flow, which requires multiple iterations until all conditions are met. Your issue will be added to our backlog for further investigation.

**Comment by Tim-Pohlmann** [comment]
The issue is caused by the combination of two issues:
1. The loop limit is unknown to the Symbolic Execution engine. While a `readonly` field cannot change, the engine does not learn this currently.
2. The SE engine does not very well handle for-loops with unknown limits.
Fixing either of these would solve this specific test case. But we should tackle both issues because they also cause problems on their own.

**Comment by Tim-Pohlmann** [comment]
I close this as a duplicate of #8028 and #8445
